{"version":1,"ops":[{"type":1,"author":{"id":"dc64d6d260e2683565fe6e13d60cfbe9fec0fa09"},"timestamp":1426200188,"metadata":{"github-id":"MDU6SXNzdWU2MDkyNTAwMw==","github-url":"https://github.com/rurban/perl-compiler/issues/44","origin":"github"},"title":"B::CC uses value from void context in next list context","message":"Original [issue 42](https://code.google.com/p/perl-compiler/issues/detail?id=42) created by rurban on 2010-08-29T10:32:10.000Z:\n\nWhile working on issue 39, I found a similar issue.\nThis test program fails with B::CC.\n# test9.pl\n\nsub f1 { 1 }\nsub f2 {\n    f1();\n    do { 7; 2 }\n}\n# print f2(), \u0026quot;\\n\u0026quot;;\n\nExpected output:\n$ perl test9.pl\n2\n\nWith B::CC:\n$ perlcc -O -o test9 test9.pl \u0026amp;\u0026amp; ./test9\n12\n\nExplanation:\n- f1 is called, it puts value 1 on the stack.\n- f2 should discard this value, because f1 is called in void context.\n- But if a block follows, this value is accidently added to the list of return values of f2.\n\nCurrently I don't know if if this is really a new issue or only another test case for issue 39.\n\nUsed versions:\n- perl v5.10.1 (*) built for i486-linux-gnu-thread-mult\n  (from Ubuntu 10.04)\n- B::C r586 from repository.","files":null}]}